src/test/java/com/neo4j/docker/TestDockerComposeSecrets.java

```diff
@@ -0,0 +1,113 @@
+package com.neo4j.docker;
+
+import com.neo4j.docker.utils.DatabaseIO;
+import com.neo4j.docker.utils.TemporaryFolderManager;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Objects;
+
+import static java.lang.System.getenv;
+
+public class TestDockerComposeSecrets
+{
+    private static final int DEFAULT_BOLT_PORT = 7687;
+    private static final int DEFAULT_HTTP_PORT = 7474;
+
+    @RegisterExtension
+    public static TemporaryFolderManager temporaryFolderManager = new TemporaryFolderManager();
+
+    private DockerComposeContainer createContainer( File composeFile, Path containerRootDir, String serviceName )
+    {
+        var container = new DockerComposeContainer( composeFile );
+
+        container.withExposedService( serviceName, DEFAULT_BOLT_PORT ).withExposedService( serviceName, DEFAULT_HTTP_PORT,
+                                                                                           Wait.forHttp( "/" ).forPort( DEFAULT_HTTP_PORT )
```

Comment:
I'm not sure this wait strategy will work across all versions and editions we test. 
There are a bunch of canned common wait strategies you can use in the [WaitStrategies.java](https://github.com/neo4j/docker-neo4j/blob/17dd78c568d34b079f6de3837e926fa89c6aae11/src/test/java/com/neo4j/docker/utils/WaitStrategies.java#L43) utility which will be more reliable.

The wait strategy should be on the bolt port service as well. Bolt should always work. IIRC http relies on the browser being bundled.